monte_carlo.py

```diff
@@ -23,60 +27,229 @@
+                
+            #test xmin
+            r = xmind/(sin(theta)*cos(phi))
+            if (0 <= r and r < neutron_distance and \
+                    r < tempd and not r == -0.0):
+                tempd = r
+                lim_bound = []
+                lim_bound.append('x_min')
+            elif r == tempd:
+                lim_bound.append('x_min')
+            
+            #test ymax
+            r = ymaxd/(sin(theta)*sin(phi))
+            if (0 <= r and r < neutron_distance and \
+                    r < tempd and not r == -0.0):
+                tempd = r
+                lim_bound = []
+                lim_bound.append('y_max')
+            elif r == tempd:
+                lim_bound.append('y_max')               
+                
+            #test ymin
+            r = ymind/(sin(theta)*sin(phi))
+            if (0 <= r and r < neutron_distance and \
+                    r < tempd and not r == -0.0):
+                tempd = r
+                lim_bound = []
+                lim_bound.append('y_min')
+            elif r == tempd:
+                lim_bound.append('y_min')               
+            
+            #test zmax
+            r = zmaxd/cos(theta)
+            if (0 <= r and r < neutron_distance and \
+                    r < tempd and not r == -0.0):
+                tempd = r
+                lim_bound = []
+                lim_bound.append('z_max')
+            elif r == tempd:
+                lim_bound.append('z_max')              
+                
+            #test zmin
+            r = zmind/cos(theta)
+            if (0 <= r and r < neutron_distance and \
+                    r < tempd and not r == -0.0):
+                tempd = r
+                lim_bound = []
+                lim_bound.append('z_min')
+            elif r == tempd:
+                lim_bound.append('z_min')               
+                
+            # if the neutron stays within the bounding box
+            if lim_bound == []:
+                neutron_distance = 0
+            neutron_position.sadd(neutron_movement_unit_vector * tempd)
+
+            for surface in lim_bound:  
+                # if the neutron is reflected
+                if (bounds.get_surface_type(surface) == 1):
+                    if surface == 'x_max' or surface == 'x_min':
+                        neutron_movement_unit_vector.smul([-1, 1, 1])
+                    if surface == 'y_max' or surface == 'y_min':
+                        neutron_movement_unit_vector.smul([1, -1, 1])
+                    if surface == 'z_max' or surface == 'z_min':
+                        neutron_movement_unit_vector.smul([1, 1, -1])
+                    neutron_reflected = True
+    
+                # if the neutron escapes
+                elif (bounds.get_surface_type(surface) == 0):
+                    neutron_lost = True
+                    neutron_interaction = 1
+                    neutron_distance = 0
+                    tallies["leaks"].increment(1)
```

Comment:
Not necessarily a huge deal, but we should stick to a convention of ``'`` or ``"``. While they are equivalent, it's nice to choose one or the other (unless the other is needed). Normally people prefer ``'``